src/dawn/sleep.cc

```diff
@@ -0,0 +1,42 @@
+// Copyright 2018 The Amber Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/dawn/sleep.h"
+
+#include "src/dawn/platform.h"
+
+#if AMBER_PLATFORM_WINDOWS
```

Comment:
What about, instead of doing ifdefs, we make a sleep_win.cc and a sleep_posix.cc and then have cmakelist make the decision of which to build? Removes the need for all the preprocessor gunk, platform.h and makes the code easier to read. ....

Reply:
Doing it in the compiled source means the cross-compile scenario is handled automatically.  Most of the ugliness is contained to the platform.h and the system-specific functions themselves.